src/main.lua

```diff
@@ -1827,6 +1818,20 @@ end, function(msg)
 	msg.ioEvent:addField("Joined-Gateways-Count", gwStats.joined)
 	msg.ioEvent:addField("Leaving-Gateways-Count", gwStats.leaving)
 	addSupplyData(msg.ioEvent)
+
+	-- Send a single tick notice to the sender after all epochs have been ticked
+	Send(msg, {
+		Target = msg.From,
+		Action = "Tick-Notice",
+		Data = json.encode({
+			distributedEpochIndexes = distributedEpochIndexes,
+			newEpochIndexes = newEpochIndexes,
+			newDemandFactors = newDemandFactors,
+			newPruneGatewaysResults = newPruneGatewaysResults,
+			tickedRewardDistributions = tickedRewardDistributions,
```

Comment:
Consider omitting this if it's not needed. Concerns over there being too much data in here (but we should think about how that could be the case in other critical scenarios).